agent/lib/internal_address_source.js

```diff
@@ -79,6 +79,22 @@ function same_allocation(a, b) {
     return true;
 }
 
+function same_messages(a, b) {
+    for (var i in a) {
+        var equal = false;
+        for (var j in b) {
+            if (a[i] === b[j]) {
+                equal = true;
+                break;
+            }
+        }
```

Comment:
Inner loop could be simplified with:
if (!b.includes(a[i])) {
  return false;
}

Reply:
Done.